Jenkinsfile

```diff
@@ -71,31 +74,49 @@ pipeline {
               return env.BUILD_FILEBEAT != "false"
             }
           }
-          stages {
-            stage('Filebeat oss'){
-              steps {
-                withBeatsEnv(){
-                  makeTarget("Filebeat oss Linux", "-C filebeat testsuite")
-                }
-              }
+          steps {
+            makeTarget("Filebeat oss Linux", "-C filebeat testsuite")
+          }
+        }
+        stage('Filebeat x-pack'){
+          agent { label 'ubuntu && immutable' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return env.BUILD_FILEBEAT != "false"
             }
-            stage('Filebeat x-pack'){
-              steps {
-                withBeatsEnv(){
-                  makeTarget("Filebeat x-pack Linux", "-C x-pack/filebeat testsuite")
-                }
-              }
+          }
+          steps {
+            makeTarget("Filebeat x-pack Linux", "-C x-pack/filebeat testsuite")
+          }
+        }
+        stage('Filebeat Mac OS X'){
+          agent { label 'macosx' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return env.BUILD_FILEBEAT != "false"
             }
-            stage('Filebeat Mac OS X'){
-              agent { label 'macosx' }
-              options { skipDefaultCheckout() }
-              steps {
-                withBeatsEnv(){
-                  makeTarget("Filebeat oss Mac OS X", "TEST_ENVIRONMENT=0 -C filebeat testsuite")
-                }
-              }
+          }
+          steps {
+            makeTarget("Filebeat oss Mac OS X", "TEST_ENVIRONMENT=0 -C filebeat testsuite")
+          }
+        }
+        stage('Filebeat Windows'){
+          agent { label 'windows-immutable && windows-2019' }
+          options { skipDefaultCheckout() }
+          when {
+            beforeAgent true
+            expression {
+              return env.BUILD_FILEBEAT != "false" && params.windowsTest
             }
           }
+          steps {
+            mageTargetWin("Filebeat oss Windows Unit test", "-d filebeat goUnitTest")
+            mageTargetWin("Filebeat oss Windows Integration test", "-d filebeat goIntegTest")
```

Comment:
Since we are just getting started I would only try to run the `goUnitTest` target on Windows for all the beats. The `goIntegTest` target needs to use Docker and none of this has been used on Windows before.

Reply:
```suggestion
            //mageTargetWin("Filebeat oss Windows Integration test", "-d filebeat goIntegTest")
```